scipy/stats/_variation.py

```diff
@@ -0,0 +1,211 @@
+        standard deviation is ``N - ddof``, where ``N`` is the number of
+        elements.  `ddof` must be less than ``N``; if it isn't, the result
+        will be ``nan`` or ``inf``, depending on ``N`` and the values in
+        the array.  By default `ddof` is zero for backwards compatibility,
+        but it is recommended to use ``ddof=1`` to ensure that the sample
+        standard deviation is computed as the square root of the unbiased
+        sample variance.
+    keepdims : bool, optional
+        If this is set to True, the axes which are reduced are left in the
+        result as dimensions with size one. With this option, the result
+        will broadcast correctly against the input array.
+
+    Returns
+    -------
+    variation : ndarray
+        The calculated variation along the requested axis.
+
+    References
+    ----------
+    .. [1] Zwillinger, D. and Kokoska, S. (2000). CRC Standard
+       Probability and Statistics Tables and Formulae. Chapman & Hall: New
+       York. 2000.
+
+    Examples
+    --------
+    >>> from scipy.stats import variation
+    >>> variation([1, 2, 3, 4, 5], ddof=1)
+    0.5270462766947299
+
+    Compute the variation along a given dimension of an array that contains
+    a few ``nan`` values:
+
+    >>> x = np.array([[  10.0, np.nan, 11.0, 19.0, 23.0, 29.0, 98.0],
+    ...               [  29.0,   30.0, 32.0, 33.0, 35.0, 56.0, 57.0],
+    ...               [np.nan, np.nan, 12.0, 13.0, 16.0, 16.0, 17.0]])
+    >>> variation(x, axis=1, ddof=1, nan_policy='omit')
+    array([1.05109361, 0.31428986, 0.146483  ])
+
+    """
+    # Import here to avoid circular import.
+    from .stats import _chk_asarray, _contains_nan
+
+    a, axis = _chk_asarray(a, axis)
+    axis = normalize_axis_index(axis, ndim=a.ndim)
+    n = a.shape[axis]
+
+    contains_nan, nan_policy = _contains_nan(a, nan_policy)
+    if contains_nan and nan_policy == 'omit':
+        return _nanvariation(a, axis=axis, ddof=ddof, keepdims=keepdims)
+
+    if a.size == 0 or ddof > n:
+        # Handle as a special case to avoid spurious warnings.
+        # The return values, if any, are all nan.
+        shp = list(a.shape)
+        if keepdims:
+            shp[axis] = 1
+        else:
+            del shp[axis]
+        if len(shp) == 0:
+            result = np.nan
+        else:
+            result = np.full(shp, fill_value=np.nan)
+
+        return result
+
+    mean_a = a.mean(axis, keepdims=True)
+
+    if ddof == n:
+        # Another special case.  Result is either inf or nan.
+        std_a = a.std(axis=axis, ddof=0)
+        result = np.full_like(std_a, fill_value=np.nan)
+        result[std_a > 0] = np.sign(mean_a) * np.inf
+        if result.shape == ():
+            result = result[()]
+        return result
+
+    std_a = a.std(axis, ddof=ddof, keepdims=True)
+
+    with np.errstate(divide='ignore', invalid='ignore'):
+        result = std_a / mean_a
```

Comment:
this context manager would ignore the division by zero error. A user could (reasonably IMO) call this on data which has zero mean, not realize, and assume errors are propogated, investigate and get confused. Would it be better to explicitly state the behavior of division by 0 mean in the docstring (in a single sentence)?

Reply:
Yes, good idea.  I'll add a comment about what happens when the mean is 0, and in the special case where the input is all zeros (in which case the return value is `nan`).

Reply:
Done.